### PR TITLE
Avoid filtering against absolute path in ignore filters

### LINKED ISSUE #2457: [Bug]: CLI not uploading files upon save
### Please confirm that you:

- [X] Are working with Shopify CLI 2.0, not 3.0. (You can [open a 3.0 issue here](https://github.com/Shopify/cli/issues/new/choose).)
- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

Upon saving a file, the change would be uploaded to the development theme (while in `shopify theme serve`)

### Actual behavior

Nothing actually happens, no upload of the change happens.

### Stack trace

_No response_

### Reproduction steps

1. Run `shopify theme serve`
2. Make a change in a file and save
3. Check the terminal to see if it shows if uploaded, as well as the theme within preview to see if the change it there

### Operating System

macOS Montery (M2 Chip)

### CLI version (run `shopify version` if you're not sure)

2.20.0

### Shell

fish

### Ruby version (run `ruby -v` if you're not sure)

3.0.0

### What language and version are you using in your application?

Node v18.5.0

### PR DESCRIPTION
**Pull request by ulgut**
### WHY are these changes introduced?

Fixes [#2457](https://github.com/Shopify/shopify-cli/issues/2457)

As seen in the issue, a developer had an issue with the syncer's ignore filter matching the absolute path of their theme files against the contents of their `.shopifyignore`. Their root theme directory was named `shopify-webpack` and the `.shopifyignore` file contained a line: `webpack/*` which matched against this root directory name.

### WHAT is this pull request doing?

This PR changes the `ignore_helper.rb`'s path matching from using absolute to relative paths.

### How to test your changes?

1. Create a theme with an absolute path name that matches some pattern inside the `.shopifyignore` file. Maybe doing `src/github.com/Shopify/testing_theme` and having `src/*` inside your `.shopifyignore`.
2. Checkout `main` and run `shopify theme serve`. Edit a file and notice that no file changes are being recognized.
3. Checkout `switch-absolute-to-relative-paths-for-filters` and rerun `shopify theme serve`. Edit a file and notice that changes are being recognized.

### Post-release steps

Follow up with developer to ensure they are seeing no further issues.

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [x] I've included any post-release steps in the section above (if needed).

### CODE REVIEW

**Review by karreiro — COMMENTED** [review]
Thank you, @jessetuglu! Very nice investigation 🚀 

I've notice that we're maybe missing a corner case. Because the file is synced, but the hot reload still ignores it.

I've slightly changed the `PathMatcher` module with these debugging logs:

```ruby
def regex_match?(regex, path)
  a = regex.match?(path)
  puts "#{a} ====> regex_match? #{regex} (#{path})"
  a
rescue StandardError
  false
end

def glob_match?(glob, path)
  a = !!::File.fnmatch?(glob, path)
  puts "#{a} ====> glob_match? #{glob} (#{path})"
  a
end
```

Here's what's being logged on the `main` branch when I change the `sections/announcement-bar.liquid` file:
Screenshot 2022-08-10 at 14 29 13

And, here's what's being logged on the `switch-absolute-to-relative-paths-for-filters` branc hwhen I change the `sections/announcement-bar.liquid` file:
Screenshot 2022-08-10 at 14 28 23

We may notice that we're still matching something in the last line, and also that nothing appears on the Google Chrome console.

Ohh.. also here's my `.shopifyignore` file:
```
src/*
```

I believe we're close to the final fix tho ☺️ - really great stuff with this investigation :)

File: lib/shopify_cli/theme/dev_server/hot_reload.rb [hunk 1]
```diff
@@ -40,14 +40,15 @@ def close
...
             .map { |file| @theme[file] }
+            .reject { |file| @ignore_filter&.ignore?(file.relative_path) }
+
 
           files -= liquid_css_files = files.select(&:liquid_css?)
 
           deleted_files = removed
-            .reject { |file| @ignore_filter&.ignore?(file) }
             .map { |file| @theme[file] }
+            .reject { |file| @ignore_filter&.ignore?(file.relative_path) }
```
Comment (jessetuglu) [inline lib/shopify_cli/theme/dev_server/hot_reload.rb:51]: same as above

**Review by mgmanzella — APPROVED** [review]
noice, testing looked great! 🚀

**Review by karreiro — APPROVED** [review]
Thank you, @jessetuglu! Great stuff 🚀

**Pull request merged**